### PR TITLE
Porting stock_picking_package_preparation* to 9

### PR DESCRIPTION
**Pull request by eLBati**
Modules
- stock_picking_package_preparation
- stock_picking_package_preparation_line

### CODE REVIEW

**Comment by gurneyalex** [comment]
rebuilding on runbot.

@eLBati did you check #212 and #226 ?

**Comment by gurneyalex** [comment]
travis should be green once #227 is merged.

**Comment by jgrandguillaume** [comment]
LGTM :+1:

**Comment by eLBati** [comment]
Hi @OpenCode 
I included your last changes to `stock_picking_package_preparation_line` to this PR.
But local tests give me

```
2016-05-13 15:44:35,913 28110 ERROR demo_9 openerp.addons.stock_picking_package_preparation_line.tests.test_package_preparation_line: `   File "/home/elbati/workspace/odoo/instances/demo9-community/parts/stock-logistics-workflow/stock_picking_package_preparation_line/tests/test_package_preparation_line.py", line 176, in test_standard_flow_with_detail_with_lot
2016-05-13 15:44:35,913 28110 ERROR demo_9 openerp.addons.stock_picking_package_preparation_line.tests.test_package_preparation_line: `     self.preparation.pack_operation_ids[0].lot_id.id,
2016-05-13 15:44:35,913 28110 ERROR demo_9 openerp.addons.stock_picking_package_preparation_line.tests.test_package_preparation_line: ` AttributeError: 'stock.pack.operation' object has no attribute 'lot_id'
```

I think this is because 9.0 introduced `stock.pack.operation.lot` model.
Could you give me any hint?

Thanks

**Comment by OpenCode** [comment]
@eLBati I don't know. I don't use Odoo 9 and I don't know it very well. Sorry.

**Comment by eLBati** [comment]
@OpenCode no problem.

Lots handling and related tests should be fixed with https://github.com/OCA/stock-logistics-workflow/pull/199/commits/2def0c34a72e77efe9099efaa1f75f978b5457cf

**Comment by eLBati** [comment]
Hello @gurneyalex , why the 'needs fixing' label?

**Comment by gurneyalex** [comment]
@eLBati runbot was red. I've triggered a rebuild.

**Comment by gurneyalex** [comment]
You need to update the version numbers in the **openerp**.py files (s/8.0/9.0/)

**Comment by eLBati** [comment]
@gurneyalex thanks, done

**Comment by gurneyalex** [comment]
@eLBati still can't figure out why runbot is red :crying_cat_face:

**Comment by eLBati** [comment]
@gurneyalex just rebuilded runbot but `stock_picking_package_preparation` is not installable.
How can I help?

**Comment by gurneyalex** [comment]
@eLBati what I just checked in #254 is that if your branch is merged with 9.0 then the CI tests are green. This can happen since the runbot does not merge, it really tests the branch so if you fork from a red build, you can get a red build on the PR. 

I'm :+1: on this, and I think we can move on and merge this.

File: stock_picking_package_preparation/model/stock_picking_package_preparation.py
Comment (lasley) [inline stock_picking_package_preparation/model/stock_picking_package_preparation.py]: api.one is deprecated, please remove all instances of it for api.multi

File: stock_picking_package_preparation/security/package_preparation_security.xml
Comment (lasley) [inline stock_picking_package_preparation/security/package_preparation_security.xml]: `<odoo noupdate="1">` & can remove data tag

File: stock_picking_package_preparation_line/README.rst
Comment (lasley) [inline stock_picking_package_preparation_line/README.rst:32]: Please use new ReadMe template https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: stock_picking_package_preparation_line/__init__.py
Comment (lasley) [inline stock_picking_package_preparation_line/__init__.py]: Please replace the old headers with the [new short one](https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3)

File: stock_picking_package_preparation_line/models/ir_config.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/ir_config.py]: Please separate model by file

File: stock_picking_package_preparation_line/models/ir_config.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/ir_config.py]: Class name should be UpperCamelCase version of model.name

File: stock_picking_package_preparation_line/models/ir_config.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/ir_config.py]: Please use new api

File: stock_picking_package_preparation_line/models/stock.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock.py]: Please name this file `stock_move`

File: stock_picking_package_preparation_line/models/stock.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock.py]: `@api.model`

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py]: required?

Also, this string attr duplicates `move_id` string. On purpose?

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py:21]: required?

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py]: `product_uom_id` - all relation fields should end in _id or _ids, depending on whether they are x2m (_ids) or x2o (_id).

This also needs a string attribute

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py:56]: `@api.model`

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py]: Please either move this into its own file (preferred), or name the file `stock_picking_package_preparation.py`

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py]: `api.model`

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py]: Please use Python docblock styling instead of C style comments: `"""` vs `#`

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py]: Please use Python docblock styling instead of C style comments: `"""` vs `#`

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py]: Are all the `-----`s necessary? They dirty the code IMO

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py]: Please use Python docblock styling instead of C style comments: `"""` vs `#`

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (lasley) [inline stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py]: Warning should be renamed as UserError per pylint-odoo. Admittedly don't know the reasoning, but I assume a planned change in the future

```
R8101   Import Warning should be renamed as UserError from openerp.exceptions import Warning as UserError
```

File: stock_picking_package_preparation_line/security/package_preparation_line_security.xml
Comment (lasley) [inline stock_picking_package_preparation_line/security/package_preparation_line_security.xml]: `<odoo noupdate="1">` & can remove data tag

File: stock_picking_package_preparation_line/views/ir_config_view.xml
Comment (lasley) [inline stock_picking_package_preparation_line/views/ir_config_view.xml:2]: `<odoo>` & remove data tag

File: stock_picking_package_preparation_line/views/stock_picking_package_preparation_line.xml
Comment (lasley) [inline stock_picking_package_preparation_line/views/stock_picking_package_preparation_line.xml]: `<odoo>` & remove data tag

**Review by lasley — CHANGES_REQUESTED** [review]
Thanks for the submission @eLBati - comments inline.

Please also rebase onto the current 9.0. If I am understanding the comments correctly, this should clear up the Runbot 🔴

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (OpenCode) [reply stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py]: This is my style to comment :)

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (eLBati) [reply stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py:21]: It's not, because `stock.picking.package.preparation.line` can be descriptive, without being linked to a stock move

File: stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py
Comment (lasley) [reply stock_picking_package_preparation_line/models/stock_picking_package_preparation_line.py]: Understood - I won't block over this, but I recommend removing. They add nothing to the code other than more maintenance. 

I am blocking on the Doc block comments though. Doc blocks are described in [PEP8](https://www.python.org/dev/peps/pep-0008/#documentation-strings) & [PEP257](https://www.python.org/dev/peps/pep-0257/)

File: stock_picking_package_preparation/model/stock_picking_package_preparation.py
Comment (archetipo) [inline stock_picking_package_preparation/model/stock_picking_package_preparation.py:107]: @eLBati 
package.get_content()  return a recordsets, is possible to do:
preparation.quant_ids = package.get_content()
?

Comment (eLBati) [reply stock_picking_package_preparation/model/stock_picking_package_preparation.py:107]: @archetipo in 9.0 `get_content` returns a list of IDs

**Comment by eLBati** [comment]
@lasley thanks, I made the changes

**Review by lasley — APPROVED** [review]

**Comment by lasley** [comment]
No clue what's up with Runbot

```
stock_picking_package_preparation_line: Unmet dependencies: stock_picking_package_preparation
```

But stock_picking_package_preparation is definitely set to installable 😑 

Code looks good though 👍

**Comment by lasley** [comment]
Once we get another signoff on this (particularly the 🔴  runbot), would you mind squashing the logical commits too please?

**Comment by pedrobaeza** [comment]
Maybe it's a runbot glitch... Let's merge and see the final result, but please first squash a bit the commits in a logical way.

**Comment by eLBati** [comment]
@lasley @pedrobaeza rebased and squashed

**Pull request merged**
